### PR TITLE
Add crosssdk support for kirkstone build

### PR DESCRIPTION
**Pull request by raennor**
### Version(s) of meta-rust
latest master

### Version(s) of poky and/or oe-core
kirkstone

### Expected result
Building of toolchain which includes nativesdk-python3-cryptography  should be finished without error

### Actual result
Build failed with following error:
```
| warning: build_rust: Forced Rust target `x86_64-pokysdk-linux` is not compatible with deduced Rust target `x86_64-unknown-linux-gnu` - the built package  may not import successfully once installed.
|
| error: Error loading target specification: Could not find specification for target "x86_64-pokysdk-linux". Run `rustc --print target-list` for a list of built-in targets
```

### Fix for this
To fix this issue I added the rust-crosssdk for all rust versions which are supported on this layer.

I also notice that librsvg 2.52.10 building failed (which is also referred on https://github.com/meta-rust/meta-rust/issues/449), which should be fixed with my second commit.

### CODE REVIEW

File: recipes-gnome/librsvg/librsvg/0002-allow-invalid-ref-casting-to-allow-building.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
+--- librsvg-2.52.10/src/surface_utils/mod.rs	2024-10-15 20:59:07.374334093 +0200
++++ librsvg-2.52.10/src/surface_utils/mod.rs_mod	2024-10-15 20:58:56.391383870 +0200
+@@ -147,12 +147,12 @@
+     #[inline]
+     fn set_pixel(&mut self, stride: usize, pixel: Pixel, x: u32, y: u32) {
+         let value = pixel.to_u32();
++        let index = (y as usize + stride) + (x as usize * 4);
```
Comment (srwalter) [inline recipes-gnome/librsvg/librsvg/0002-allow-invalid-ref-casting-to-allow-building.patch:7]: Shouldn't this be `y as usize * stride`?

File: recipes-gnome/librsvg/librsvg/0002-allow-invalid-ref-casting-to-allow-building.patch [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ 
+-        #[allow(clippy::cast_ptr_alignment)]
+-        unsafe {
+-            let p: *mut u8 = &mut self[y as usize * stride + x as usize * 4];
+-            *(p as *mut u32) = value;
+-        }
++        assert!(index + 4 <= self.len(), "Index out of bounds");
```
Comment (srwalter) [inline recipes-gnome/librsvg/librsvg/0002-allow-invalid-ref-casting-to-allow-building.patch:14]: This check is both incorrect and unnecessary.  copy_from_slice will do bounds checking.

**Review by srwalter — CHANGES_REQUESTED** [review]

File: recipes-gnome/librsvg/librsvg/0002-allow-invalid-ref-casting-to-allow-building.patch (hunk 1 above)
Comment (raennor) [reply recipes-gnome/librsvg/librsvg/0002-allow-invalid-ref-casting-to-allow-building.patch:7]: Oh, yes you are absolutely correct, this was a typo.

File: recipes-gnome/librsvg/librsvg/0002-allow-invalid-ref-casting-to-allow-building.patch (hunk 2 above)
Comment (raennor) [reply recipes-gnome/librsvg/librsvg/0002-allow-invalid-ref-casting-to-allow-building.patch:14]: Thanks for this information. I didn't know that copy_from_slice do bounds checking, so I will remove this check.

**Comment by srwalter** [comment]
Does your patch also apply to the version of librsvg in mickledore?  Consider using librsvg_%.bbappend as the name, if so

**Comment by raennor** [comment]
Hi @srwalter,
sorry, but this patch will only apply until Version 2.53.1 because this function **DerefMut** was removed in version 2.53.2.

**Comment by raennor** [comment]
Hi @srwalter,
I have modified the bbappend so that it will be applied also to mickledore (librsvg_%.bbappend ) but the patch and  and the TARGET_CC_ARCH will only be expanded if PV is 2.52.10 (which is present in kirkstone).

This should possibly fix the workflow issue on mickledore.

**Review by srwalter — APPROVED** [review]

**Pull request merged**